src/sentry/grouping/api.py

```diff
@@ -67,14 +68,18 @@ class GroupingConfigLoader:
 
     cache_prefix: str  # Set in subclasses
 
-    def get_config_dict(self, project: Project) -> GroupingConfig:
+    def get_config_dict(self, project: Project, detector: Detector | None = None) -> GroupingConfig:
```

Comment:
it seems like we're passing around the detector a bit here - is there anything we can do to reduce the blast radius (diff size)? Do we need to pass it to both get_config_id and get enhancements? could we add another method that takes the detector info an handles all this?